src/Microsoft.Data.Sqlite/SqliteFactory.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Data.Common;
+
+namespace Microsoft.Data.Sqlite
+{
+    public class SqliteFactory : DbProviderFactory
+    {
+        static SqliteFactory()
+        {
+            Instance = new SqliteFactory();
+        }
+
+        private SqliteFactory()
+        {
+        }
+
+        public static SqliteFactory Instance { get; }
```

Comment:
Oh yuck, I just noticed [the other ones](https://msdn.microsoft.com/en-us/library/vstudio/system.data.sqlclient.sqlclientfactory.instance) are ***fields***. We should probably conform in case something in .Full NET depends on that.

For a bit more context, you can register these things in `Web.config`...
```XML
<system.data>
  <DbProviderFactories>
    <add invariant="Bubblegum"
         type="Microsoft.Data.Sqlite.SqliteFactory" />
  </DbProviderFactories>
</system.data>
```
...then load them from code.
```C#
var factory = DbProviderFactories.GetFactory("Bubblegum");
```